### PR TITLE
fixes #9405 - Attaches custom products when registering with activation key

### PR DESCRIPTION
**Pull request by cfouant**

### CODE REVIEW

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-subscriptions.html [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+          formatter="booleanToYesNo"
+          readonly="denied('edit_activation_keys', activationKey)"
+          on-save="save(activationKey)">
+    </span>
+  </div>
+  <p bst-alert="info" ng-hide="!activationKey.auto_attach">
+    <span translate>When Auto Attach is enabled, the minimal amount of Red Hat subscriptions on the activation key which are required to validate the machine will be attached upon registration. All custom products will be attached.</span>
+  </p>
+  <p bst-alert="info" ng-hide="activationKey.auto_attach">
+    <span translate>When Auto Attach is disabled, every possible subscription on the activation key will be attached upon registering a machine.</span>
```
Comment (jlsherrill) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-subscriptions.html]: I'd say something like:

When Auto Attach is disabled, registering systems will be attached to all associated subscriptions.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-subscriptions.html [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+    <span class="info-label" translate>Auto-Attach</span>
+    <span class="info-value"
+          bst-edit-checkbox="activationKey.auto_attach"
+          formatter="booleanToYesNo"
+          readonly="denied('edit_activation_keys', activationKey)"
+          on-save="save(activationKey)">
+    </span>
+  </div>
+  <p bst-alert="info" ng-hide="!activationKey.auto_attach">
+    <span translate>When Auto Attach is enabled, the minimal amount of Red Hat subscriptions on the activation key which are required to validate the machine will be attached upon registration. All custom products will be attached.</span>
```
Comment (jlsherrill) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-subscriptions.html]: The wording is a little awkward, but i'm having trouble coming up with anything better?  Something like:

When Auto Attach is enabled, registering systems will be attached to all associated custom products and only associated Red Hat subscriptions required to satisfy the systems installed Products. 


even though its not thats still not great.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-subscriptions.html [hunk 3]
```diff
@@ -1,5 +1,25 @@
 <span page-title ng-model="activationKey">{{ 'Subscriptions for Activation Key:' | translate }} {{ activationKey.name }}</span>
 
+<section>
+  <h4 translate>Activation Key Type:</h4>
+
+  <div class="detail">
```
Comment (jlsherrill) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-subscriptions.html]: I think this class should be 'details'.  It makes the 'editable' interaction not span the entire width.

**Comment by jlsherrill** [comment]
Otherwise looks good to me!

**Comment by cfouant** [comment]
Alright @jlsherrill - changes made, let me know what you think. Thanks!

**Comment by jlsherrill** [comment]
ACK

**Pull request merged**